addon/components/search-result/component.js

```diff
@@ -0,0 +1,174 @@
+import Ember from 'ember';
+import layout from './template';
+import Analytics from '../../mixins/analytics';
+import hostAppName from '../../mixins/host-app-name';
+
+/**
+ * Adapted from Ember-SHARE and Ember Preprints
+ * Used for search results on discover page.
+ *
+ * ```handlebars
+ * {{search-result
+ *      detailRoute=detailRoute
+ *      addFilter='addFilter'
+ *      result=result
+ *      queryParams=queryParams
+ *      filterReplace=filterReplace
+ *      updateFilters=(action 'updateFilters')
+ * }}
+ * ```
+ * @class search-result
+ */
+export default Ember.Component.extend(Analytics, hostAppName, {
+    layout,
+    maxTags: 10,
+    maxSubjects: 10,
+    maxCreators: 10,
+    maxDescription: 300,
+    showBody: false,
+    /**
+     * Array of query params being used in consuming app
+     * @property {Array} queryParams
+     */
+    queryParams: null,
+    /**
+     * Search result from SHARE
+     * @property {Object} result
+     */
+    result: null,
+    providerUrlRegex: {
+        //'bioRxiv': '', doesnt currently have urls
+        Cogprints: /cogprints/,
+        OSF: /https?:\/\/((?!api).)*osf.io/, // Doesn't match api.osf urls
+        PeerJ: /peerj/,
+        arXiv: /arxivj/,
+        'ClinicalTrials.gov': /http:\/\/clinicaltrials.gov/,
+    },
+    /**
+     * Name of detail route for consuming application, if you want search result to link to a route in the consuming spp
+     * @property {String} detailRoute
+     */
+    detailRoute: null,
+    footerIcon: Ember.computed('showBody', function() {
+        return this.get('showBody') ? 'caret-up' : 'caret-down';
+    }),
+
+    type: Ember.computed('result.type', function() {
+        return this.get('result.type').replace(/\w\S*/g, function(str) {return str.capitalize();});
+    }),
+    safeTitle: Ember.computed('result.title', function() {
+        return Ember.String.htmlSafe(this.get('result.title')).string;
+    }),
+    safeDescription: Ember.computed('result.description', function() {
+        return Ember.String.htmlSafe(this.get('result.description')).string;
```

Comment:
Will be removed.  Related ticket here https://openscience.atlassian.net/browse/SHARE-755